test/jdk/javax/swing/JComboBox/DisabledComboBoxFontTestAuto.java

```diff
@@ -0,0 +1,176 @@
+        for (int x = (enabledImage.getWidth() / 2) - 20;
+             x < (enabledImage.getWidth() / 2) + 20; x++) {
+            // Nimbus has a pixel offset in coordinates since Nimbus is 2px
+            // smaller in width than other L&F's
+            if (lafName.equals("Nimbus")) {
+                eColor1 = new Color(enabledImage.getRGB(x + 1, y));
+                eColor2 = new Color(enabledImage2.getRGB(x, y));
+                dColor1 = new Color(disabledImage.getRGB(x + 1, y));
+                dColor2 = new Color(disabledImage2.getRGB(x, y));
+            } else {
+                eColor1 = new Color(enabledImage.getRGB(x, y));
+                eColor2 = new Color(enabledImage2.getRGB(x, y));
+                dColor1 = new Color(disabledImage.getRGB(x, y));
+                dColor2 = new Color(disabledImage2.getRGB(x, y));
+            }
+            if ((!isColorMatching(eColor1, eColor2)) || (!isColorMatching(dColor1, dColor2))) {
+                isIdentical = false;
+                break;
+            }
+        }
+
+        if (isIdentical) {
+            System.out.println("PASSED");
+        } else {
+            throw new RuntimeException("FAIL: " +
+                    "Enabled and disabled ComboBox matches");
+        }
+    }
+
+    private static boolean isColorMatching(Color c1, Color c2) {
+        if ((c1.getRed() != c2.getRed())
+                || (c1.getBlue() != c2.getBlue())
+                || (c1.getGreen() != c2.getGreen())) {
+
+            System.out.println(lafName + " Enabled RGB failure: "
+                    + c1.getRed() + ", "
+                    + c1.getBlue() + ", "
+                    + c1.getGreen() + " vs "
+                    + c2.getRed() + ", "
+                    + c2.getBlue() + ", "
+                    + c2.getGreen());
+            return false;
+        }
+        return true;
+    }
+
+    private static void setLookAndFeel(UIManager.LookAndFeelInfo laf) {
+        try {
+            UIManager.setLookAndFeel(laf.getClassName());
+        } catch (UnsupportedLookAndFeelException ignored){
+            System.out.println("Unsupported LookAndFeel: " + laf.getClassName());
+        } catch (ClassNotFoundException | InstantiationException |
+                IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static void main(String[] args) throws Exception {
+        lafName = "null";
+        testDir = Path.of(System.getProperty("test.classes", "."));
+        for (UIManager.LookAndFeelInfo laf : UIManager.getInstalledLookAndFeels()) {
+            lafName = laf.getName().equals("CDE/Motif") ? "Motif" : laf.getName();
+            System.out.println("LAF is " + lafName);
+            SwingUtilities.invokeAndWait(() -> setLookAndFeel(laf));
+            SwingUtilities.invokeAndWait(DisabledComboBoxFontTestAuto::createCombo);
+            SwingUtilities.invokeAndWait(DisabledComboBoxFontTestAuto::paintCombo);
+            testMethod();
```

Comment:
So test will fail whenever it encounters the first wrong color? Ideally we would run test for all the installed LAFs anyways and report which ones are failed. But in this case you would need to pass the LAF name to the test method so it saves screenshots for all the LAFs for future analysis.

Reply:
I added a StringBuffer which now outputs the failed LAFs to the thrown runtime exception.